src/bun.js/node/types.zig

```diff
@@ -1835,548 +1849,2876 @@ pub const Emitter = struct {
 };
 
 pub const Path = struct {
+    const CHAR_BACKWARD_SLASH = '\\';
+    const CHAR_COLON = ':';
+    const CHAR_DOT = '.';
+    const CHAR_FORWARD_SLASH = '/';
+    const CHAR_QUESTION_MARK = '?';
+
+    const CHAR_STR_BACKWARD_SLASH = "\\";
+    const CHAR_STR_FORWARD_SLASH = "/";
+    const CHAR_STR_DOT = ".";
+
+    const StringBuilder = @import("../../string_builder.zig");
+
+    // tmpBuf1, tmpBuf2, and tmpBuf3 can be expanded by path methods when
+    // they encounter larger strings.
+    var _tmpBuf1: PathBuffer = undefined;
+    var _tmpBuf2: PathBuffer = undefined;
+    var _tmpBuf3: PathBuffer = undefined;
+
+    var tmpBuf1: []u8 = &_tmpBuf1;
+    var tmpBuf2: []u8 = &_tmpBuf2;
+    var tmpBuf3: []u8 = &_tmpBuf3;
+
+    var smallTmpBuf1: PathBuffer = undefined;
```

Comment:
fyi these are not small on windows (96kb)

Reply:
Ya, really term is fixed or expandable buffers. I'll noodle on naming.